src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java

```diff
@@ -246,197 +279,222 @@ Optional<DatasetProperty> extract(DatasetRegistrationSchemaV1 registration, Cons
         }
     };
 
-    // TODO: is there some way to do this automatically from, e.g., the schema?
-    private static final List<DatasetPropertyExtractor> DATASET_REGISTRATION_V1_PROPERTY_EXTRACTORS = List.of(
-            new DatasetPropertyExtractor(
-                    "PI Name", "piName", PropertyType.String,
-                    (registration, consentGroup) -> registration.getPiName()),
-            new DatasetPropertyExtractor(
-                    "Study Name", "studyName", PropertyType.String,
-                    (registration, consentGroup) -> registration.getStudyName()),
-            new DatasetPropertyExtractor(
-                    "Study Type", "studyType", PropertyType.String,
-                    (registration, consentGroup) -> {
+    public record StudyPropertyExtractor(
```

Comment:
Can we pull this out into its own class?

Reply:
I am honestly not super happy with this approach. I want to rethink it - it's overly complex just to save some lines of code. I feel like we should keep it as-is and reconsider later refactoring it. I'll put in a comment to do that as a follow up, but I feel like it's low-priority (personally).